runtime/runtime/src/verifier.rs

```diff
@@ -112,46 +112,54 @@ pub fn set_tx_state_changes(
     set_account(state_update, tx.signer_id().clone(), &signer);
 }
 
-/// Verifies the signed transaction on top of the given state; looks up the
-/// signer account and access_key from the transaction; updates them to charge
-/// for the transaction processing; returns the updated signer and access_key to
-/// the caller.  The caller can use `commit_charging_for_tx()` to commit the
-/// actual charging.
-pub fn verify_and_charge_tx_ephemeral(
-    config: &RuntimeConfig,
+pub fn get_singer_and_access_key(
     state_update: &TrieUpdate,
     validated_tx: &ValidatedTransaction,
-    transaction_cost: &TransactionCost,
-    block_height: Option<BlockHeight>,
-) -> Result<VerificationResult, InvalidTxError> {
-    let _span = tracing::debug_span!(target: "runtime", "verify_and_charge_transaction").entered();
-
-    let TransactionCost { gas_burnt, gas_remaining, receipt_gas_price, total_cost, burnt_amount } =
-        *transaction_cost;
+) -> Result<(Account, AccessKey), InvalidTxError> {
+    let signer_id = validated_tx.signer_id();
 
-    let tx = validated_tx.to_tx();
-    let signer_id = tx.signer_id();
-
-    let mut signer = match get_account(state_update, signer_id)? {
+    let signer = match get_account(state_update, signer_id)? {
         Some(signer) => signer,
         None => {
             return Err(InvalidTxError::SignerDoesNotExist { signer_id: signer_id.clone() });
         }
     };
 
-    let mut access_key = match get_access_key(state_update, signer_id, tx.public_key())? {
+    let access_key = match get_access_key(state_update, signer_id, validated_tx.public_key())? {
         Some(access_key) => access_key,
         None => {
             return Err(InvalidTxError::InvalidAccessKeyError(
                 InvalidAccessKeyError::AccessKeyNotFound {
                     account_id: signer_id.clone(),
-                    public_key: tx.public_key().clone().into(),
+                    public_key: validated_tx.public_key().clone().into(),
                 },
             )
             .into());
         }
     };
+    Ok((signer, access_key))
+}
 
+/// Verifies the provided signed transaction against the given signer account and access key.
+/// Updates the signer account balance and increments the access key nonce to account for transaction processing costs.
+/// Returns an error if verification fails due to nonce issues, insufficient balance, access key permission violations,
+/// or storage stake requirements. On success, returns the result detailing gas usage and costs,
+/// allowing the caller to subsequently commit the charges via `commit_charging_for_tx`.
```

Comment:
```suggestion

```


I recommend getting rid of these comments.  They are outdated and show that we are not able to keep them up-to-date.